packages/fuels-accounts/src/predicate.rs

```diff
@@ -98,8 +98,11 @@ impl ViewOnlyAccount for Predicate {
         self.address()
     }
 
-    fn try_provider(&self) -> AccountResult<&Provider> {
-        self.provider.as_ref().ok_or(AccountError::no_provider())
+    fn try_provider(&self) -> Result<&Provider> {
+        self.provider.as_ref().ok_or(error!(
+            Other,
+            "no provider available. Make sure to use `set_provider`"
```

Comment:
We have a copy of this message in 3 places. Can we have it as a helper on `ViewOnlyAccount`